wallets/provider-ledger/src/signers/ethereumSigner/ledgerEthereumSigner.ts

```diff
@@ -0,0 +1,81 @@
+import type { TransactionLike } from 'ethers';
+import type { GenericSigner } from 'rango-types';
+import type { EvmTransaction } from 'rango-types/lib/api/main';
+
+import AppEth, { ledgerService } from '@ledgerhq/hw-app-eth';
+import TransportU2F from '@ledgerhq/hw-transport-webhid';
+import { JsonRpcProvider, Transaction } from 'ethers';
+import { SignerError } from 'rango-types';
+
+import { ETH_BIP32_PATH, getLedgerErrorMessage } from '../../helpers';
+
+export const RPC_PROVIDER_URL = 'https://rpc.ankr.com/eth';
+
+export class LedgerEthereumSigner implements GenericSigner<EvmTransaction> {
+  async signMessage(): Promise<string> {
+    // TODO: Should be implemented using eth.signPersonalMessage
+    throw SignerError.UnimplementedError('signMessage');
+  }
+
+  async signAndSendTx(
+    tx: EvmTransaction,
+    fromAddress: string,
+    chainId: string | null
+  ): Promise<{ hash: string }> {
+    let transport;
+    try {
+      const provider = new JsonRpcProvider(RPC_PROVIDER_URL); // Provider to broadcast transaction
+
+      const transactionCount = await provider.getTransactionCount(fromAddress); // Get nonce
+
+      const transaction: TransactionLike<string> = {
+        to: tx.to,
+        gasPrice: tx.gasPrice,
+        gasLimit: tx.gasLimit,
+        nonce: transactionCount,
+        chainId: chainId,
+        data: tx.data,
+        value: tx.value,
+        maxPriorityFeePerGas: tx.maxPriorityFeePerGas,
+        maxFeePerGas: tx.maxFeePerGas,
+      };
+
+      const unsignedTx =
+        Transaction.from(transaction).unsignedSerialized.substring(2); // Create unsigned transaction
+
+      const resolution = await ledgerService.resolveTransaction(
```

Comment:
Regarding this document: https://developers.ledger.com/docs/connectivity/ledgerJS/clear-signing

Should we pass second and third params as well? we are passing empty object.

This maybe helpful as well:
https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/hw-app-eth/tests/ledgerService.unit.test.ts

Reply:
Those two objects are empty because all the properties of those objects are optional but they are not optional themselves. I checked those properties and didn't find anything that we need to pass. They include properties like nftExplorerBaseURL (Backend service responsible for signed NFT APDUS), pluginBaseURL, extraPlugins (extra plugins to add for the resolution), cryptoassetsBaseURL., domains (List of trusted names to clear sign)

Reply:
How about `erc20` option? what it does?

Reply:
ResolutionConfig Allows to configure precisely what the service need to resolve. It seems we can set erc20:true if we need clear signing on ERC20 transaction. If we set it and it is not an ERC20 transaction, it should still work but will do a useless service resolution. In most of examples I see that every parameter is set to true like this: 

`const resolutionConfig: ResolutionConfig = {
  nft: true,
  erc20: true,
  externalPlugins: true,
`};``